units/generate_tasks.go

```diff
@@ -294,6 +280,12 @@ func (j *generateTasksJob) Run(ctx context.Context) {
 		"job":           j.ID(),
 		"version":       t.Version,
 	}))
+
+	if err != nil && !shouldNoop {
+		j.AddError(err)
+		return
```

Comment:
I may be missing something, as I haven't read the code in some time, but what about the case where a generator legitimately errors, e.g., if there is a syntax error in the JSON? In that case we should still mark the task as having generated tasks.

Reply:
That depends on what we want the flag to mean exactly. Just from the name, I would argue that it should still be false since the generator did not successfully run. For intermittent errors it's probably better that we allow a subsequent execution to try again. For non-intermittent errors it's unclear, but I would think that doing some redundant work to come to the same error is probably ok

Reply:
Sorry for the delayed reply. Which flag are you referring to? What concerns me is that with this change we might continue to retry even though there is a legitimate error.

Reply:
I was talking about the GeneratedTasks flag in the task data structure. My philosophy behind the changes here is that if you restart the task and we run the generate.tasks command again, it's ok to lean more toward rerunning the generation logic, because presumably it would error in the same way if there's a legitimate error and would only do so once (and is arguably the right thing to do because someone requested that it run again)